src/lib/libwebaudio.js

```diff
@@ -301,6 +301,14 @@ var LibraryWebAudio = {
     return emscriptenGetContextQuantumSize(contextHandle);
   },
 
+  emscripten_audio_context_sample_rate: (contextHandle) => {
+#if ASSERTIONS
+    assert(EmAudio[contextHandle], `Called emscripten_audio_context_sample_rate() with an invalid Web Audio Context handle ${contextHandle}`);
+    assert(EmAudio[contextHandle] instanceof (window.AudioContext || window.webkitAudioContext), `Called emscripten_audio_context_sample_rate() on handle ${contextHandle} that is not an AudioContext, but of type ${EmAudio[contextHandle]}`);
+#endif
```

Comment:
How about adding a `getAudioContext(handle)` helper that can take care of adding these assertions to all APIs that need to lookup a handler like this?

Then this would just be `emscripten_audio_context_sample_rate: (handle) => getAudioContext(handler)['sampleRate'],`

Reply:
> How about adding a `getAudioContext(handle)` helper that can take care of adding these assertions to all APIs that need to lookup a handler like this?

That would add to code size in non-ASSERTIONS builds. I think it's nice to see the flat assertions like this, since it helps with not having to read deep into nested sub-functions.